bottle.py

```diff
@@ -550,14 +550,14 @@ def prepare(self):
 
     def all_plugins(self):
         """ Yield all Plugins affecting this route. """
-        unique = set()
-        for p in reversed(self.app.plugins + self.plugins):
-            if True in self.skiplist: break
-            name = getattr(p, 'name', False)
-            if name and (name in self.skiplist or name in unique): continue
-            if p in self.skiplist or type(p) in self.skiplist: continue
-            if name: unique.add(name)
-            yield p
+        if True in self.skiplist: return
+        skips = set(self.skiplist)
+        for plugins in self.plugins, self.app.plugins:
+            for p in reversed(plugins):
```

Comment:
Compared to the original code I find this much harder to understand.
normally a `for` loop in python is in the form:
```
for single in plural
    do something ... 
```
You wrote for `plugins` in `<other list of plugins>`. I didn't dive into analyzing in depth to see what is the other set of plugins. But this form of `for` will immediately cause a headache for a lot of people.
Further, an embedded for loop is harder to understand compared to a simple for loop.